server/src/main/java/org/opensearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java

```diff
@@ -148,7 +148,7 @@ protected Aggregator createInternal(
             : includeExclude.convertToStringFilter(DocValueFormat.RAW, maxRegexLength);
 
         MapStringTermsAggregator.CollectorSource collectorSource = new SignificantTextCollectorSource(
-            queryShardContext.lookup().source(),
+            new SourceLookup(),
```

Comment:
`sourceLookup` is final in `SearchLookup`, so without changing that we can't just create a new `sourceLookup` for each `SearchLookup` -- also `SearchLookup` is also shared between threads so that wouldn't solve our problem anyways. On the Aggregator side `SignificantTextAggregatorFactory` is the only place where `SearchLookup::source` is used, so in the aggregator side it doesn’t look like we need to access `SourceLookup` through `SearchLookup` again so it shouldn't matter that we aren't getting `SourceLookup` from `SearchLookup` here.

Reply:
may be lets add a comment here: `creating separate SourceLookup per slice is fine as it is owned by this Agg instance and not accessed anywhere else`